### PR TITLE
Add rpm support in integTest framework (2nd PR)

### LINKED ISSUE #1951: Automate integ test using RPM bundle.
Currently our integ tests are working only for using tarball distribution. We now have workflow to generate RPM bundle. We need incorporate the automated integ tests for RPM bundle.

### PR DESCRIPTION
**Pull request by peterzhuamazon**
Signed-off-by: Peter Zhu <zhujiaxi@amazon.com>

### Description
Add rpm support in integTest framework (2nd PR)
 
### Issues Resolved
#1951
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by peterzhuamazon** [comment]
Hi @dblock please review as I will add test cases soon.

Thanks.

File: src/test_workflow/integ_test/distribution_rpm.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+                '&&',
+                'yum',
+                'install',
+                '-y',
+                bundle_name
+            ]
+        )
+        subprocess.check_call(rpm_install_cmd, cwd=self.work_dir, shell=True)
+
+    @property
```
Comment (dblock) [inline src/test_workflow/integ_test/distribution_rpm.py:44]: You could subclass this further and get rid of all the maps. So `DistributionTar` becomes `DistributionOpenSearchTar` and ``DistributionOpenSearchDashboardsTar`. Then all these classes have different `start_cmd` properties that just returns the right value. You could get rid of `filename` altogether then. 

The other option is to keep `filename` and generate everything from it. So `start_cmd` is `f"systemctl start {filename}"` (or some transform of filename)

**Review by dblock — CHANGES_REQUESTED** [review]
This is a lot better, but go further in subclassing or generalize the distribution start/stop code. Basically get rid of those maps of product -> commands.

File: src/test_workflow/integ_test/distribution.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    @abstractmethod
+    def get_start_cmd(self) -> str:
+        """
+        Return the start command for the distribution
+        """
+        pass
+
+    def cleanup(self) -> None:
+        """
+        Allow distribution that is not 'tar' to do proper cleanup
```
[the remark this answers is not in the document]
Comment (peterzhuamazon) [reply src/test_workflow/integ_test/distribution.py:52]: Tar does not need to do it as tarball will be put in tmp folder and delete after the run complete.

[the remark this answers is not in the document]
Comment (peterzhuamazon) [reply src/test_workflow/integ_test/distribution.py:52]: Cleanup does not necessarily means uninstall, can also mean clean up additional files that is getting generated not as part of the original artifact. I think it is better than just uninstall.

If you strong suggest it I can rename it to uninstall.

Thanks.

File: src/test_workflow/integ_test/distribution_rpm.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    def get_install_dir(self) -> str:
+        return os.path.join(os.sep, "usr", "share", self.filename)
+
+    @property
+    def get_config_dir(self) -> str:
+        return os.path.join(os.sep, "etc", self.filename)
+
+    def install_distribution(self, bundle_name: str) -> None:
```
[the remark this answers is not in the document]
Comment (peterzhuamazon) [reply src/test_workflow/integ_test/distribution_rpm.py:26]: My distribution class is relying on service class to send all the information.
I dont see any issues tho but if you think there is better way please let me know.
Thanks.

File: src/test_workflow/integ_test/distribution_rpm.py (hunk 1 above)
Comment (peterzhuamazon) [reply src/test_workflow/integ_test/distribution_rpm.py:44]: The problem is in `tar` we can not generate everything as the start command for os and osd are completely different without any patterns.

My confusion is would we go this further to split classes this small, just to not use a map?

Thanks.

File: src/test_workflow/integ_test/distribution.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    @property
+    @abstractmethod
+    def get_config_dir(self, bundle_name: str) -> str:
+        """
+        Return the config directory for the distribution
+        """
+        pass
+
+    @abstractmethod
+    def install_distribution(self) -> None:
```
[the remark this answers is not in the document]
Comment (peterzhuamazon) [reply src/test_workflow/integ_test/distribution.py]: Renamed.
Thanks.

File: src/test_workflow/integ_test/distribution.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (peterzhuamazon) [reply src/test_workflow/integ_test/distribution.py:52]: Renamed.
Thanks.

File: src/test_workflow/integ_test/distribution.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        """
+        The detailed method to install the distribution before start the service
+        """
+        pass
+
+    @property
+    @abstractmethod
+    def get_start_cmd(self) -> str:
```
[the remark this answers is not in the document]
Comment (peterzhuamazon) [reply src/test_workflow/integ_test/distribution.py]: Renamed.
Thanks.

File: src/test_workflow/integ_test/distribution.py [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        return self
+
+    def __init__(self, filename: str, version: str, work_dir: str) -> None:
+        self.filename = filename
+        self.version = version
+        self.work_dir = work_dir
+
+    @property
+    @abstractmethod
+    def get_install_dir(self) -> str:
```
[the remark this answers is not in the document]
Comment (peterzhuamazon) [reply src/test_workflow/integ_test/distribution.py]: Renamed.
Thanks.

File: src/test_workflow/integ_test/distribution_rpm.py (hunk 1 above)
Comment (peterzhuamazon) [reply src/test_workflow/integ_test/distribution_rpm.py:44]: After some further checking I think I will keep it this way.
With 90% of the code to be exactly the same and only 10% differs, it is hard for me to think separating to more subclasses would benefit the code.

Tho I would definitely get rid of one map on rpm, as rpm is very similar between os and osd, not tar.

Thanks.

**Comment by peterzhuamazon** [comment]
@dblock @tianleh @zelinh please review again.

Thanks.

**Pull request merged**